api/types/kustomization.go

```diff
@@ -318,17 +316,8 @@ func (k *Kustomization) EnforceFields() []string {
 
 // Unmarshal replace k with the content in YAML input y
 func (k *Kustomization) Unmarshal(y []byte) error {
-	j, err := yaml.YAMLToJSON(y)
-	if err != nil {
-		return err
+	if err := yaml.UnmarshalStrict(y, &k); err != nil {
+		return fmt.Errorf("kustomization unmarshal error: %w", err)
```

Comment:
Since the message from `err` itself already mentions unmarshalling, I suggest changing this to 
```suggestion
		return errors.WrapPrefixf(err, "invalid Kustomization")
```
(and using `sigs.k8s.io/kustomize/kyaml/errors` for wrapping)

Reply:
fix